Sources/ServiceContextModule/ServiceContext.swift

```diff
@@ -251,15 +244,16 @@ extension ServiceContext {
     /// To access the task-local value, use `ServiceContext.current`.
     ///
     /// SeeAlso: [Swift Task Locals](https://developer.apple.com/documentation/swift/tasklocal)
-    #if swift(>=5.7)
-    @_unsafeInheritExecutor // same as withValue declared in the stdlib; because we do not want to hop off the executor at all
-    public static func withValue<T>(_ value: ServiceContext?, operation: () async throws -> T) async rethrows -> T {
+#if swift(>=6.0)
+    public static func withValue<T>(_ value: ServiceContext?,
```

Comment:
Actually after looking at this again this is an API break strictly speaking. We need both API shapes in Swift 6 as well since somebody could reference the method. We can just disfavour the old one

Reply:
I have mixed feelings about this rule tbh